### PR TITLE
Remove Phusion base image; Change tagging strategy

### PR DESCRIPTION
**Pull request by marek-jakubowski**
### Desired Outcome

*Goal is to provide some changes from #112 to make it smaller and deliver tagging strategy earlier.*

### Implemented Changes

*Describe how the desired outcome above has been achieved with this PR. In
particular, consider:*

- _What's changed? Why were these changes made?_
    Introduced new tagging strategy
    Removed Phusion image in favour of specific image for appliance (eg. with Runit)
- _How should the reviewer approach this PR, especially if manual tests are required?_
- _Are there relevant screenshots you can add to the PR description?_

### Connected Issue/Story

Resolves #[relevant GitHub issue(s), e.g. 76]

CyberArk internal issue ID: [insert issue ID]

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [x] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

**Review by micahlee — COMMENTED** [review]
I know you're still working on this, so I'll hold off on approving yet until you're ready. This is looking great, though!

I just had a couple of small notes on the changes. Thanks!

File: ubuntu-ruby-fips/setuser/main.go
Comment (marek-jakubowski) [reply ubuntu-ruby-fips/setuser/main.go:13]: Originally in Phusion image there was bash script for doing that, @tarnowsc wrote Go code for that. I'll provide proper README with information what is this for.

File: ubuntu-ruby-fips/Dockerfile
Comment (marek-jakubowski) [reply ubuntu-ruby-fips/Dockerfile:107]: This will be used as replacement for phusion-ruby-fips (only know location for me is in the appliance). There are more occurrences but mostly could be replaced with lighter ubuntu-ruby-builder just for build-essentials and ruby (for example for 'bundle install').
Actually our work targets to remove from conjur/appliance need of installing additional tools. In that case base images would take care of installing everything they need, and then in appliance there would be not install command.

Comment (micahlee) [reply ubuntu-ruby-fips/Dockerfile:107]: Hmm... okay. I'll let this be for now, then.

I'm not sure if I'm a fan of intentionally separating strongly coupled parts of the appliance where there isn't re-use sharing taking place. This goes against another effort we've been slowly undertaking to consolidate more of the appliance into a single repo, so that there's a single place to understand and change the appliance.

That's not something to resolve in this PR or project, though! 😅

Comment (jvanderhoof) [reply ubuntu-ruby-fips/Dockerfile:107]: I agree with your concern @micahlee. I'd like to see us be very specific with the dependencies and where we apply them.  From the list below, I'd recommend we install `keyutils`, `slapd`, `ldap-utils`, `runit`, `libarchive13`, `cron`,  `syslog-ng-core`, and `syslog-ng-mod-sql` in the appliance as these are all dependencies specific to the appliance.  The rest of these dependencies are import for both Conjur and Conjur Enterprise, and thus, make sense to manage here.

Comment (marek-jakubowski) [reply ubuntu-ruby-fips/Dockerfile:107]: I assume that `ubuntu-ruby-postgres-fips` will be used only by appliance in it's Dockerfile.
We planned to deliver all needed dependencies in base-image repo and then in destination repo (for example Conjur or appliance) use that image only to configure and prepare application.

**Review by micahlee — APPROVED** [review]
LGTM! Thanks for the effort iterating on this one, @marek-jakubowski. It's turned out really solid! 😄

**Comment by marek-jakubowski** [comment]
Push-forced with some documentation improvement (forgot to remove Phusion mentions). @micahlee please check it again.

**Review by codihuston — APPROVED** [review]
Re-approving after the updated docs. Thanks!

**Pull request merged**
